addons/sale_management/models/sale_order.py

```diff
@@ -71,49 +78,51 @@ def update_prices(self):
 
     @api.onchange('sale_order_template_id')
     def onchange_sale_order_template_id(self):
+
         if not self.sale_order_template_id:
             self.require_signature = self._get_default_require_signature()
             self.require_payment = self._get_default_require_payment()
             return
+
         template = self.sale_order_template_id.with_context(lang=self.partner_id.lang)
 
+        # --- first, process the list of products from the template
         order_lines = [(5, 0, 0)]
         for line in template.sale_order_template_line_ids:
             data = self._compute_line_data_for_template_change(line)
+
             if line.product_id:
+                price = line.product_id.lst_price
                 discount = 0
+
                 if self.pricelist_id:
-                    price = self.pricelist_id.with_context(uom=line.product_uom_id.id).get_product_price(line.product_id, 1, False)
-                    if self.pricelist_id.discount_policy == 'without_discount' and line.price_unit:
-                        discount = (line.price_unit - price) / line.price_unit * 100
-                        # negative discounts (= surcharge) are included in the display price
-                        if discount < 0:
```

Comment:
Is this chunk of code useless ? Do we really prevent a pricelist to set a higher price than the lst_price ? 

If not, I think that it should be reintroduced.